pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -387,15 +387,20 @@ func (c *Reconciler) createBuildPod(ctx context.Context, tr *v1alpha1.TaskRun, t
 
 	// For each step with no entrypoint set, try to populate it with the info
 	// from the remote registry
-	cache, err := entrypoint.NewCache()
+	entrypointCache, err := entrypoint.NewCache()
 	if err != nil {
 		return nil, fmt.Errorf("couldn't create new entrypoint cache: %v", err)
 	}
+	digestCache, err := entrypoint.NewCache()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create new digest cache: %v", err)
+	}
 	bSpec := bs.DeepCopy()
 	for i := range bSpec.Steps {
 		step := &bSpec.Steps[i]
 		if len(step.Command) == 0 {
-			ep, err := entrypoint.GetRemoteEntrypoint(cache, step.Image)
+			digest, err := entrypoint.GetImageDigest(digestCache, step.Image)
+			ep, err := entrypoint.GetRemoteEntrypoint(entrypointCache, digest)
```

Comment:
nice, im glad we do this in two steps 👍

Reply:
@Zyqsempai Actually shouldn't `entrypoint.GetRemoteEntrypoint` take `step.Image` *and* `digest` ?
The *digested` image should be in the form of `step.Image@digest` otherwise the ref is wrong :sweat_smile: